kernel.py

```diff
@@ -152,9 +159,15 @@ def _write_connection_file(self):
         # The key should be secret, to only allow the same user to connect.
         # Make sure the permissions are set accordingly.
         os.chmod(self.connection_filename, os.stat(self.connection_filename).st_mode & 0o0700)
+
+        def shorten_filename(runtime_file):
+            """Shorten connection filename kernel-24536.json -> 24536"""
+            r_cfile = r'.*kernel-([^\-]*).*\.json'
+            return re.sub(r_cfile, r'\1', runtime_file)
+
```

Comment:
This `shorten_filename` is only correct if it used the `jupyter_runtime_dir`, right?
Also, what if `connection_filename` had another pattern?

Reply:
If there is no match no replacement.
The danger is some `../some_dir/anystuff1-kernel-anystuff2.json`. So to correct that, we must keep an other member variable to know if we are on default  conf (i.e. None)

Reply:
Yes, that's what I mean.
Btw, also, isn't it easier to just not introduce `shorten_filename` here, and instead have the code directly? It's much shorter and thus easier to understand.

Reply:
Yes inline is more readable. Finally the function got pretty short.
Prettify : c824bcf89